pennylane/tape/tape.py

```diff
@@ -349,6 +365,42 @@ def __exit__(self, exception_type, exception_value, traceback):
         finally:
             QuantumTape._lock.release()
 
+    @property
+    def circuit(self):
+        """Returns the quantum circuit recorded by the tape.
+
+        The circuit is created with the assumptions that:
+
+        * The ``operations`` attribute contains quantum operations and
+          mid-circuit measurements and
+        * The ``measurements`` attribute contains terminal measurements.
+
+        Note that the resulting list could contain MeasurementProcess objects
+        that some devices may not support.
+
+        Returns:
+
+            list[.Operator, .MeasurementProcess]: the quantum circuit
+            containing quantum operations and measurements as recorded by the
+            tape.
+        """
+        return self.operations + self.measurements
+
+    def __iter__(self):
+        """list[.Operator, .MeasurementProcess]: Return an iterator to the
+        underlying quantum circuit object."""
+        return iter(self.circuit)
+
+    def __getitem__(self, idx):
+        """list[.Operator]: Return the indexed operator from underlying quantum
+        circuit object."""
+        return self.circuit[idx]
+
+    def __len__(self):
+        """int: Return the number of operators in the underlying quantum
```

Comment:
Might be better to say the sum of number of operations and observables?

Reply:
+1